google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/readrows/ReadRowsBatchingDescriptor.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.data.v2.stub.readrows;
+
+import com.google.api.core.SettableApiFuture;
+import com.google.api.gax.batching.BatchingDescriptor;
+import com.google.api.gax.batching.BatchingRequestBuilder;
+import com.google.cloud.bigtable.data.v2.models.Query;
+import com.google.cloud.bigtable.data.v2.models.Row;
+import com.google.common.base.Preconditions;
+import com.google.protobuf.ByteString;
+import java.util.List;
+
+// TODO: JavaDoc
+public class ReadRowsBatchingDescriptor
+    implements BatchingDescriptor<ByteString, Row, Query, List<Row>> {
+
+  @Override
+  public BatchingRequestBuilder<ByteString, Query> newRequestBuilder(Query query) {
+    return new BulkReadRowsRequestBuilder(query);
+  }
+
+  @Override
+  public void splitResponse(List<Row> rows, List<SettableApiFuture<Row>> futureResults) {
+    // TODO: verify if this check
+    Preconditions.checkState(rows.size() == futureResults.size());
```

Comment:
This needs to be a bit more careful:
- you need to consider cases when keys don’t exist
- when there are duplicate keys

Which means that we need to able to match the original requested key to the row key. 

I need to take a closer look, but it seems like this will require changes to the gax BatchingDescriptor api...it needs to access the original request entries.

Reply:
Also this needs unit tests to handle those cases